services/blockstorage/adapter/filesystem/file_system_persistence.go

```diff
@@ -52,12 +52,20 @@ type FilesystemBlockPersistence struct {
 	codec        blockCodec
 }
 
-func NewBlockPersistence(ctx context.Context, conf config.FilesystemBlockPersistenceConfig, parent log.Logger, metricFactory metric.Factory) (adapter.BlockPersistence, error) {
-	logger := parent.WithTags(log.String("adapter", "block-storage"))
+func (f *BlockPersistence) GracefulShutdown(shutdownContext context.Context) {
+	if err := f.blockWriter.Close(); err != nil {
+		f.logger.Error("failed to close blocks file")
+		return
+	}
+	f.logger.Info("closed blocks file")
+}
+
+func NewBlockPersistence(conf config.FilesystemBlockPersistenceConfig, parent log.Logger, metricFactory metric.Factory) (*BlockPersistence, error) {
+	logger := parent.WithTags(log.String("adapter", "block-storage"), log.String("filename", blocksFileName(conf)))
```

Comment:
this logger is stored in the adapter struct. is the filename really needed in every line of log?